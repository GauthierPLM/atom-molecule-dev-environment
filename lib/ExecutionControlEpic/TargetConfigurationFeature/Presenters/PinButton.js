'use babel'
// @flow

import React from 'react';

export default class PinButton extends React.Component<DefaultProps, Props, State> {

  state: State;
  props: Props;
  static defaultProps: DefaultProps;

  constructor(props: Props) {
    super(props);
  }

  render() {
    return (
      <button className="btn icon icon-pin" onClick={e => {e.stopPropagation(); return this.props.onClick()}}/>
    )
  }
}

PinButton.propTypes = {

};

PinButton.defaultProps = {

};

type DefaultProps = {

};

type Props = {
  onClick(): void,
};

type State = {

};
