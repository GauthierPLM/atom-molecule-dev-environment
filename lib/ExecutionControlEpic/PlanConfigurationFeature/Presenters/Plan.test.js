import React from 'react';
import {shallow, mount} from 'enzyme';
import Plan from "./Plan";
import PinButton from './PinButton';
import {StyleRoot} from 'radium';

describe('Plan', () => {
  it('should display the plan name', () => {
    let subject = mount(<StyleRoot><Plan name="build"/></StyleRoot>);

    expect(subject.findWhere(comp => comp.type() == "span" && comp.text() == "build").length).toBe(1);
  });

  it('should display a pin button when pinnable props is passed', () => {
    let subject = mount(<StyleRoot><Plan pinnable/></StyleRoot>);

    expect(subject.find(PinButton).length).toBe(1);
  });

  it('should display the tool icon', () => {
    let subject = mount(<StyleRoot><Plan iconUri="atom://myplugin/icon.png"/></StyleRoot>);

    expect(subject.find('img').at(0).prop('src')).toBe('atom://myplugin/icon.png');
  });

  it('should call onPin when pin button is clicked', () => {
    let spy = jest.fn();
    let subject = mount(<StyleRoot><Plan pinnable onPin={spy}/></StyleRoot>);

    subject.find(PinButton).at(0).simulate('click');

    expect(spy).toBeCalled();
  });
});
