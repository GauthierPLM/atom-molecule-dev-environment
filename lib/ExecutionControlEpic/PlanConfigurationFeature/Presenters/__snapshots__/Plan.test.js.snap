// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plan should display a context menu with Pin and Remove when pinnable is passed 1`] = `
Array [
  MyMenuItem {
    "click": [Function],
    "label": "Pin",
  },
  MyMenuItem {
    "click": [Function],
    "label": "Remove",
  },
]
`;

exports[`Plan should display a context menu with Unpin and Remove when pinnable is not passed 1`] = `
Array [
  MyMenuItem {
    "click": [Function],
    "label": "Un-pin",
  },
  MyMenuItem {
    "click": [Function],
    "label": "Remove",
  },
]
`;
